### PR TITLE
Adds .NET Core templates, invoke support

### PR DESCRIPTION
**Pull request by andyhopp**
*Issue #, if available:*

*Description of changes:*
This includes additional project templates along with the changes to the templates necessary to support invoke/mock.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-dotnet-runtime-provider/amplify-plugin.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "functionRuntime",
+  "id": "0f41f976-bb82-4651-9576-141ad9dfdf8e",
```
[the remark this answers is not in the document]
Comment (andyhopp) [reply packages/amplify-dotnet-runtime-provider/amplify-plugin.json]: Done!

File: packages/amplify-dotnet-runtime-provider/amplify-plugin.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  "type": "util",
+  "commands": [],
+  "eventHandlers": [],
+  "functionRuntime": {
+    "pluginId": "amplify-dotnet-function-runtime-provider",
```
[the remark this answers is not in the document]
Comment (andyhopp) [reply packages/amplify-dotnet-runtime-provider/amplify-plugin.json]: Done!

File: packages/amplify-dotnet-runtime-provider/src/utils/build.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  });
+}
+
+function isBuildStale(resourceDir: string, lastBuildTimestamp: Date) {
+  const dirTime = new Date(fs.statSync(resourceDir).mtime);
+  if (dirTime > lastBuildTimestamp) {
+    return true;
+  }
+  const fileUpdatedAfterLastBuild = glob
+    .sync(`${resourceDir}/*/!(bin|obj)/**`)
```
[the remark this answers is not in the document]
Comment (andyhopp) [reply packages/amplify-dotnet-runtime-provider/src/utils/build.ts]: Appears to be working in the most recent version.

File: packages/amplify-dotnet-template-provider/resources/lambda/InvocationShim/InvocationShim.csproj.ejs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
[the remark this answers is not in the document]
Comment (andyhopp) [reply packages/amplify-dotnet-template-provider/resources/lambda/InvocationShim/InvocationShim.csproj.ejs]: I don’t think that will work here. I’m supporting both .NET 2.1 and 3.1, which requires that I build the shim using the correct runtime (hence the project template). While I have currently turned off 3.1, we’ll be enabling it as soon as it goes GA.

File: packages/amplify-dotnet-template-provider/resources/lambda/Serverless/event.json [hunk 5]
```diff
@@ -0,0 +1,85 @@
+{
+  "body": "eyJ0ZXN0IjoiYm9keSJ9",
+  "resource": "/{proxy+}",
+  "path": "/items",
```
[the remark this answers is not in the document]
Comment (andyhopp) [reply packages/amplify-dotnet-template-provider/resources/lambda/Serverless/event.json]: Good catch; modified the event file.

File: packages/amplify-dotnet-function-runtime-provider/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { FunctionRuntimeContributorFactory, CheckDependenciesResult } from 'amplify-function-plugin-interface';
+import { constants } from './constants';
+import { detectDotNetCore } from './utils/detect';
+import { build } from './utils/build';
+import { packageAssemblies } from './utils/package';
+import { invoke } from './utils/invoke';
+
+export const functionRuntimeContributorFactory: FunctionRuntimeContributorFactory = (context: any) => {
+  return {
+    checkDependencies: async () => {
```
Comment (edwardfoyle) [inline packages/amplify-dotnet-function-runtime-provider/src/index.ts:10]: tested this out without dotnet installed and got:
```
? Provide the AWS Lambda function name: netTest1
? Choose the function runtime that you want to use: .NET Core 3.1
Error: spawn dotnet ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:264:19)
    at onErrorNT (internal/child_process.js:456:16)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:270:12)
    at onErrorNT (internal/child_process.js:456:16)
    at processTicksAndRejections (internal/process/task_queues.js:81:21) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn dotnet',
  path: 'dotnet',
  spawnargs: [ '--version' ]
}
 !  ~/a/i/multiRuntime                    10.6s  Wed Apr  1 12:14:46 2020
```

File: packages/amplify-dotnet-function-runtime-provider/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      const dotNetCorePresent = await detectDotNetCore();
+      var result: CheckDependenciesResult = {
+        hasRequiredDependencies: dotNetCorePresent,
+      };
+      if (!dotNetCorePresent) {
+        result.errorMessage = `Unable to find dotnet version ${constants.currentSupportedVersion} on the path.`;
```
Comment (edwardfoyle) [inline packages/amplify-dotnet-function-runtime-provider/src/index.ts:16]: provide a link to install in the error message

**Pull request merged**